mlflow/server/handlers.py

```diff
@@ -261,12 +262,62 @@ def initialize_backend_stores(backend_store_uri=None, default_artifact_root=None
     except UnsupportedModelRegistryStoreURIException:
         pass
 
+def _assert_string(x):
+    assert isinstance(x, str)
+
+def _assert_int(x):
+    assert isinstance(x, int)
+
+def _assert_bool(x):
+    assert isinstance(x, bool)
+
+def _assert_float(x):
+    assert isinstance(x, float)
+
+def _assert_array(x):
+    assert isinstance(x, list)
+
+def _assert_required(x):
+    assert x is not None
+
+def _assert_less_than_or_equal(x, max):
+    assert x <= max
+
+def _assert_item_type_string(x):
+    for item in x:
+        _assert_string(item)
+
+
+def _validate_param_against_schema(schema, param, value, parsing_succeeded = False):
```

Comment:
Nit: Can we add a docstring  to this method and document each parameter, particularly `parsing_succeeded`? For clarity, can we also rename this to `proto_parsing_succeeded`?